pkg/ebpf/events_pipeline.go

```diff
@@ -519,7 +519,7 @@ func (t *Tracee) deriveEvents(ctx context.Context, in <-chan *trace.Event) (
 					case events.PrintMemDump:
 					default:
 						// Derived events might need filtering as well
-						if t.matchPolicies(&derivative) == 0 {
+						if t.matchPolicies(&derivatives[i]) == 0 {
```

Comment:
Good catch.

What about to remove the copied var usage above (derivative) and make all accesses via index to avoid further errors like this?

```go
				for i := range derivatives {
					// Skip events that dont work with filtering due to missing types being handled.
					// https://github.com/aquasecurity/tracee/issues/2486
switch events.ID(derivatives[i].EventID) {
```

Reply:
Good idea. This will also avoid the extra copy of creating derivative instance in the loop

Reply:
> Good catch.
> 
> What about to remove the copied var usage above (derivative) and make all accesses via index to avoid further errors like that?
> 
> ```go
> 				for i := range derivatives {
> 					// Skip events that dont work with filtering due to missing types being handled.
> 					// https://github.com/aquasecurity/tracee/issues/2486
> switch events.ID(derivatives[i].EventID) {
> ```

And the code suggestion was just to get us on the same page. Please be my guest on code style or naming, nits, anything 😄.